Lib/test/test_os.py

```diff
@@ -2873,6 +2873,48 @@ def test_getfinalpathname_handles(self):
 
         self.assertEqual(0, handle_delta)
 
+    @support.requires_subprocess()
+    def test_stat_unlink_race(self):
+        # bpo-46785: the implementation of os.stat() falls back to reading
+        # the parent directory if CreateFileW() fails with a permission
+        # error. If reading the parent directory fails because the file or
+        # directory are subsequently unlinked, or because the volume or
+        # share are no longer available, then the original permission error
+        # should not be restored.
+        filename =  os_helper.TESTFN
+        self.addCleanup(os_helper.unlink, filename)
+        command = '''if 1:
+            import os
+            import sys
+            filename = sys.argv[1]
+            while True:
```

Comment:
You could also pass `deadline` in and only loop that long. Avoids the harsh termination at the end of the test.

Reply:
That works. Just wait on the child process for 5 seconds.

Reply:
Will do. By the way, is there any reason as to why arguments like `filename` and `deadline` should be passed to the script using argv, as opposed to having the script be an f-string (and using string interpolation)?

Reply:
Done (using argv).